pygmt/src/grdhisteq.py

```diff
@@ -0,0 +1,74 @@
+"""
+grdhisteq - Compute directional gradients from a grid.
+"""
+
+import xarray as xr
+from pygmt.clib import Session
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@use_alias(
+    C="divisions",
+    G="outgrid",
+    R="region",
+    N="gaussian",
+    Q="quadratic",
+    V="verbose",
+    n="interpolation",
+)
+@kwargs_to_strings(R="sequence")
+def grdhisteq(grid, **kwargs):
+    r"""
+    Perform histogram equalization for a grid.
+
+    Full option list at :gmt-docs:`grdhisteq.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    grid : str or xarray.DataArray
+        The file name of the input grid or the grid loaded as a DataArray.
+    outgrid : str or None
+        The name of the output netCDF file with extension .nc to store the grid
+        in.
+    divisions : int or str
+        The number of divisions of data range to make [Default is 16].
```

Comment:
1. The gmt docs language is confusing here, but grdhisteq defaults to -C16 if -C is not provided. But if -C is provided, *n_cells* must also be appended (e.g., this fails `gmt grdhisteq @earth_relief_05m -R0/1/0/1 -C -D`).

2. I prefer aliases that are one word rather than split, so I prefer `divisions` or `cells`. I think `equalization` is less descriptive for this parameter. What is your preference?

Reply:
>     1. The gmt docs language is confusing here, but grdhisteq defaults to -C16 if -C is not provided. But if -C is provided, _n_cells_ must also be appended (e.g., this fails `gmt grdhisteq @earth_relief_05m -R0/1/0/1 -C -D`).

Ah that's interesting, so only `int` then and no `bool`. Not sure about the `[Default is 16]` text, might get a bit confusing for users indeed.

>     2. I prefer aliases that are one word rather than split, so I prefer `divisions` or `cells`. I think `equalization` is less descriptive for this parameter. What is your preference?

To be honest, I was sort of leaning on `equalization` because `equalization=True` sounded nice. But since `bool` options (`-C`) might not work, I prefer `divisions`.